app/frontend/src/components/forms/SubmissionsTable.vue

```diff
@@ -232,13 +232,13 @@ export default {
     },
   },
 
-  mounted() {
-    // Get the form and latest form fields
-    this.fetchForm(this.formId).then(() => {
-      this.fetchFormFields({
-        formId: this.formId,
-        formVersionId: this.form.versions[0].id,
-      });
+  async mounted() {
+    // Get the form, latest version form fields
+    // These calls have to be waited on until the table can populate below
+    await this.fetchForm(this.formId);
+    await this.fetchFormFields({
+      formId: this.formId,
+      formVersionId: this.form.versions[0].id,
     });
```

Comment:
It would be worthwhile to change this to the following promise-chain pattern in order to maximize parallelization:
```js
mounted() {
  Promise.all([
    this.getFormPermissionsForUser(this.formId),
    this.fetchForm(this.formId).then(() => {
      this.fetchFormFields({
        formId: this.formId,
        formVersionId: this.form.versions[0].id,
      });
    }),
  ]).then(() = {
    this.populateSubmissionsTable();
  });
}
```
To my understanding, `populateSubmissionsTable` needs all of the above previous things to have resolved first before proceeding. However, `fetchFormFields` depends on `fetchForm`, but NOT on `getFormPermissionsForUser`. So you want to make sure that `getFormPermissionsForUser` can run independently, and NOT block `fetchFormFields` nor `fetchForm` from running at the same time.